content/2022/robot-valjak.md

```diff
@@ -2,3 +2,76 @@
+
+Transfer funkcija predstavlja odnos između signala upravljačkog sistema i ulaznog signala, za sve moguće ulazne vrednosti. Za bilo koji kontrolni sistem postoji referentni ulaz ili pobuda koji prolazi kroz transfer funkciju kako bi se proizveo odgovor sistema na odgovarajuću pobudu.
+
+Transfer funkcija (funkcija prenosa) je definisana kao Laplasova transformacija izlazne promenjljive i Laplasove transformacije ulazne promenjljive pod pretpostavkom da su svi početni uslovi jednaki nuli.
+$$G(s)\ =\ \frac{C(s)}{R(s)}$$
+Način na koji se formira transfer funkcija kontrolnog sistema je sledeća:
+
+1. Formiraju se jednačine sistema
+2. Uzimaju se Laplasove transformacije jednačina, pretpostavljajući da su početni uslovi jednaki nuli
+3. Određuju se ulazi i izlazi sistema
+4. Na kraju se uzima odnos Laplasove transformacije izlaza i ulaza što je tražena transfer funkcija
+
+Takođe, važno je naglasiti da izlaz i ulaz nekog sistema ne mora nužno da bude istih fizičkih dimenzija.
+
+U radu su korišćene tri transfer funkcije koje se uz data pravila mogu izvesti iz fizičkih jednačina za horizontalno kretanje valjka. To su transfer funkcija ugla inklinacije klatna valjka i napona na motoru, transfer funkcija ugaone brzine klatna valjka i napona na motoru i, na kraju, ugaona brzina valjka i napona motora.
+
+TODO: Analiza stabilnosti transfer funkcija
+
+##### PID simulacije
+
+Kako bi se razumeo PID kontroler potrebno je definisati odredjene termine. *Željena vrednost* je obično vrednost koju unosi korisnik. Na primer, za sistem za grejanje to bi bila željena temperatura prostorije. *Izlaz* je kontrolisana vrednost PID kontrolera. U automobilu bi to bila količina goriva koja je potrebna da bi se održavala konstantna brzina. *Greška* je vrednost koju koristi PID kontroler da bi odredio kako da manipuliše izlazom kako bi se dostigla željena vrednost. Greška se računa kao razlika željene vrednosti i trenutne vrednosti sistema.
+
+U radu su simulirana tri PID kontrolera. Prvi PID kontroler odžava konstantan ugao inklinacije klatna u odnosu na horizontalnu podlogu po kojoj se valjak kreće. Drugi PID kontroler održava konstantnu ugaonu brzinu valjka. Poslednji PID kontroler održava konstantan ugaoni pređeni put valjka. Blok dijagram kontrolnog sistema je predstavljen na slici:
+
+Kao što je u prethodnom poglavlju rečeno, transfer funkcija predstavlja odnos između signala upravljačkog sistema i ulaznog signala, za sve moguće ulazne vrednosti. Ukoliko je transfer funkcija predstavljena preko Laplasovih transformacija onda je takva funkcija prenosa definisana na kontinualnom domenu. Takav oblik transfer funkcija je od koristi prilikom teorijskih analiza kao što je analiza polova i nula. Nazalost, u praksi nam treba diskretna reprezentacija sistema što proizilazi iz toga da mikrokontroler rade isključivo nad diskretnim domenom vrednosti. Postoje razni algoritmi koji vrše željenu konverziju. Jedan od njih, koji je i koriščen u radu, *Zero-order Hold* metoda 
```

Comment:
1. Nemojte da ponavljate definicije
2. "Kao što je u prethodnom poglavlju rečeno" izbaciti (isto ako i pre komentar što sam dao)
3. "Ukoliko je transfer funkcija predstavljena preko Laplasovih transformacija onda je takva funkcija prenosa definisana na kontinualnom domenu" - Ovo je malo bezveze napisano. Bolje objasnite ovo u trenutku kada pišete o transfer funkcijama, ako vam je to bitno, pa posle pišite "kontinualna transfer funkcija" i "diskretna transfer funkcija"
4. "Takav oblik transfer funkcija je od koristi prilikom teorijskih analiza kao što je analiza polova i nula." Vi ovo niste radili. Pored toga, niste objasnili šta su polovi i nule.
5. Zašto "Nažalost"? To je dobra stvar da se diskretizuje, zato što možemo da pravimo kompleksnije kontrolere na jednostavniji način.
6. Objasnite šta je "Zero-order hold" i zašto ste ga koristili

Reply:
Stavke od jedan do pet su odradjene. Što se tiče ZOHa je ista situacija kao i za matričnu transfer funkciju, ja nigde nisam implementirana ZOH funkcija već je samo prosledjena kao parametar Control funkciji.